packages/adapter-next/src/index.ts

```diff
@@ -1,3 +1,8 @@
+import path from "path";
+
+const rootFile = import.meta.url.slice("file://".length);
+globalThis.__dirname = path.dirname(rootFile);
```

Comment:
Node.js has a standard function for this:

```ts
import { fileURLToPath } from "node:url";

fileURLToPath(import.meta.url)
```

Example:

```ts
```ts
import { fileURLToPath } from "node:url";

fileURLToPath(new URL("./relative/path/to/file", import.meta.url))
```

If possible, could we avoid adding variables to the global scope? It shouldn't be necessary in this case.

Reply:
I'm unable to use URL in adapters, for whatever reason (with an error like "self is not referenced" when initialising the plugin)

Reply:
Strange error! If you know the reason, I'm curious to know what it was.